File: wheystore/wheystore.Web/Modules/SanPhamDB/DonHang/DonHangColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.DonHang")]
[BasedOnRow(typeof(DonHangRow), CheckNames = true)]
public class DonHangColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int MaDh { get; set; }
    public int MaKhachHang { get; set; }
    [EditLink]
    public string HoTenNguoiNhan { get; set; }
    public string DiaChiNguoiNhan { get; set; }
    public string SoDienThoaiNguoiNhan { get; set; }
    public string EmailNguoiNhan { get; set; }
    public string TinhThanhNguoiNhan { get; set; }
    public string QuanHuyenNguoiNhan { get; set; }
    public string PhuongXaNguoiNhan { get; set; }
    public string MaBuuChinhNguoiNhan { get; set; }
    public int ShipperId { get; set; }
    public DateTime NgayDatHang { get; set; }
    public DateTime NgayGiaoHangDuKhien { get; set; }
    public DateTime NgayGiaoHangHoanThanh { get; set; }
    public decimal TongTienHang { get; set; }
    public decimal PhiVanChuyen { get; set; }
    public decimal TongThanhToan { get; set; }
    public decimal GiamGiaDonHang { get; set; }
    public string PhuongThucThanhToan { get; set; }
    public bool DaThanhToan { get; set; }
    public short TrangThaiDonHang { get; set; }
    public string GhiChu { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/DonHangColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DonHangRow } from "./DonHangRow";

export interface DonHangColumns {
    MaKhachHang: Column<DonHangRow>;
    HoTenNguoiNhan: Column<DonHangRow>;
    DiaChiNguoiNhan: Column<DonHangRow>;
    SoDienThoaiNguoiNhan: Column<DonHangRow>;
    EmailNguoiNhan: Column<DonHangRow>;
    ShipperId: Column<DonHangRow>;
    NgayDatHang: Column<DonHangRow>;
    NgayGiaoHangDuKhien: Column<DonHangRow>;
    NgayGiaoHangHoanThanh: Column<DonHangRow>;
    TongTienHang: Column<DonHangRow>;
    PhiVanChuyen: Column<DonHangRow>;
    TongThanhToan: Column<DonHangRow>;
    GiamGiaDonHang: Column<DonHangRow>;
    PhuongThucThanhToan: Column<DonHangRow>;
    DaThanhToan: Column<DonHangRow>;
    GhiChu: Column<DonHangRow>;
    CreatedAt: Column<DonHangRow>;
    UpdatedAt: Column<DonHangRow>;
}

export class DonHangColumns extends ColumnsBase<DonHangRow> {
    static readonly columnsKey = 'SanPhamDB.DonHang';
    static readonly Fields = fieldsProxy<DonHangColumns>();
}